crates/spirv-std/src/textures.rs

```diff
@@ -42,6 +42,28 @@ impl Image2d {
             result
         }
     }
+    /// Fetch a single texel with a sampler set at compile time
+    #[spirv_std_macros::gpu_only]
+    pub fn fetch<I, V>(&self, coordinate: V) -> Vec4
+    where
+        I: Integer,
+        V: Vector<I>,
```

Comment:
I think it's better the coordinate uses impl trait, this way you don't need to specify it if you need to set the integer type.

```suggestion
    pub fn fetch<I>(&self, coordinate: impl Vector<I>) -> Vec4
    where
        I: Integer,
```

Reply:
Makes sense, I have updated to this